src/directives/badge/badge.directive.ts

```diff
@@ -0,0 +1,170 @@
+import { Directive, ElementRef, Input, Renderer, OnInit, OnChanges, SimpleChanges } from '@angular/core';
+import { ColorService } from 'src/services/color/color.service';
+import { ThemeColor } from 'src/services/color/theme-color';
+import { ContrastService } from './../accessibility/contrast-ratio/contrast.service';
+
+export enum BadgeVerticalPosition {
+    Top = 'Top',
```

Comment:
Probably best to make all the enum string values lower case to match all our other enums